rpc/jsoncodec/conn.go

```diff
@@ -60,17 +61,41 @@ func (conn *wsJSONConn) Receive(msg interface{}) error {
 const closingIODeadline = 10 * time.Second
 
 func (conn *wsJSONConn) Close() error {
-	// Ater we close, all readers and writers must be forced to unblock.
+	c := conn.conn.NetConn()
+
+	// After we close, all readers and writers
+	// must be forced to unblock immediately.
 	defer func() {
-		c := conn.conn.NetConn()
-		_ = c.SetWriteDeadline(time.Now().Add(closingIODeadline))
-		_ = c.SetReadDeadline(time.Now().Add(closingIODeadline))
+		_ = c.SetDeadline(time.Now())
 	}()
 
 	if err := conn.writeClose(); err != nil {
```

Comment:
We made a little mistake here before. The `readClose` has to happen when `writeClose` succeeds.. not when it errors.